scriptworker/task.py

```diff
@@ -422,35 +391,59 @@ def is_try(task, source_env_prefix):
     ))
 
 
-def is_pull_request(context, task):
+async def is_pull_request(context, task):
     """Determine if a task is a pull-request-like task (restricted privs).
 
-    This goes further than checking ``tasks_for``.  We may or may not want
+    This goes further than checking ``tasks_for``. We may or may not want
     to keep this.
 
     This checks for the following things::
 
         * ``task.extra.env.tasks_for`` == "github-pull-request"
         * ``task.payload.env.MOBILE_HEAD_REPOSITORY`` doesn't come from an official repo
         * ``task.metadata.source`` doesn't come from an official repo, either
+        * The last 2 items are landed on the official repo
```

Comment:
Agreed. This implementation paves the way to https://github.com/mozilla-releng/scriptworker/issues/273. We just need to parse the bits of HTML returned by the `branch_commits` endpoint and we can know what branch the commit lives on. 

I'm not sure how we can  check if it's part of the last N commits of a branch, yet. The real github APIv3 may have something already.

Reply:
Ah, I thought it might be easier to detect the last N commits of a branch, rather than just knowing which branch(es) a commit is on. If the latter is easier, it's preferable.

Reply:
Mmm, I thought there was a way to get the last N commits of a branch, but this doesn't seem to be exposed after all:  https://developer.github.com/v3/repos/branches/

`branch_commits` is our best alternative.